### ISSUE TITLE
Google::Cloud::Vision::V1::ProductSearchClient.product_set_path raises an error for args with spaces in them

### ISSUE THREAD

**Issue by jerryskye** [googleapis/google-cloud-ruby#4291]
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

Please run down the following list and make sure you've tried the usual "quick fixes":

  - Search the issues already opened: https://github.com/googleapis/google-cloud-ruby/issues
  - Search StackOverflow: http://stackoverflow.com/questions/tagged/google-cloud-platform+ruby

If you are still having issues, please be sure to include as much information as possible:

#### Environment details

  - OS: macOS Catalina 10.15
  - Ruby version: 2.5.3
  - Gem name and version: `google-cloud-vision (0.36.0)`

#### Steps to reproduce

  1. Call `Google::Cloud::Vision::V1::ProductSearchClient`'s `product_set_path` method with a string with a space in it as any of its three arguments.

#### Code example

```ruby
require 'google/cloud/vision'
Google::Cloud::Vision::ProductSearch.new.product_set_path('one', 'two', 'three with spaces')

```

#### Full backtrace

```sh
irb(main):001:0> require 'google/cloud/vision'
=> true
irb(main):002:0> Google::Cloud::Vision::ProductSearch.new.product_set_path('one', 'two', 'three with spaces')
Traceback (most recent call last):
       13: from /Users/rubyist/.rbenv/versions/2.5.3/bin/irb:11:in `<main>'
       12: from (irb):2
       11: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-cloud-vision-0.36.0/lib/google/cloud/vision/v1/helpers.rb:1138:in `product_set_path'
       10: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-cloud-vision-0.36.0/lib/google/cloud/vision/v1/product_search_client.rb:158:in `product_set_path'
        9: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-gax-1.8.1/lib/google/gax/path_template.rb:173:in `render'
        8: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-gax-1.8.1/lib/google/gax/path_template.rb:173:in `each'
        7: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-gax-1.8.1/lib/google/gax/path_template.rb:180:in `block in render'
        6: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-gax-1.8.1/lib/google/gax/path_template.rb:180:in `new'
        5: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-gax-1.8.1/lib/google/gax/path_template.rb:135:in `initialize'
        4: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-gax-1.8.1/lib/google/gax/path_template.rb:61:in `parse'
        3: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/rly-0.2.3/lib/rly/yacc.rb:65:in `parse'
        2: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/rly-0.2.3/lib/rly/lex.rb:164:in `next'
        1: from /Users/rubyist/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/google-gax-1.8.1/lib/google/gax/path_template.rb:46:in `block in <class:PathLex>'
RuntimeError (Syntax error at ' ')

```

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

**Comment by blowmage** [comment]
The `product_set_path` helper method is intended to be used to create a `ProductSet` resource name. According to the [`ProductSet#name`](https://googleapis.dev/ruby/google-cloud-vision/latest/Google/Cloud/Vision/V1/ProductSet.html#name-instance_method) documentation, the resource name has the following format:

`projects/PROJECT_ID/locations/LOC_ID/productSets/PRODUCT_SET_ID`

The [`product_set_path` helper method](https://googleapis.dev/ruby/google-cloud-vision/latest/Google/Cloud/Vision/V1/ProductSearchClient.html#product_set_path-class_method) also takes three arguments:

* project
* location
* product_set

These arguments relate to the `PROJECT_ID`, `LOC_ID`, and `PRODUCT_SET_ID` values in the resource name format. None of these individual values, or the `ProductSet` resource name, allow spaces in them.

Why are you attempting to pass values with spaces to this method?

**Comment by jerryskye** [comment]
@blowmage Oops :blush: I didn't notice that spaces weren't permitted 😅
However please take a look at this, because I was able to create a product set named `test product set` with spaces 🤔
I used the code below to do that:
```ruby
Google::Cloud::Vision::ProductSearch.new.create_product_set(
  my_location_path,
  { display_name: 'test product set' }
)
```

Here I pull the name of my product set:
```sh
irb(main):001:0> require 'google/cloud/vision'
=> true
irb(main):002:0> product_search_client = Google::Cloud::Vision::ProductSearch.new; nil
=> nil
irb(main):003:0> location_path = product_search_client.location_path 'project-id', 'location'
=> "projects/project_id/locations/location"
irb(main):004:0> product_set = product_search_client.list_product_sets(location_path).response.product_sets.first; nil
=> nil
irb(main):005:0> product_set.name
=> "projects/project_id/locations/location/productSets/test product set"
irb(main):006:0> _.include? ' '
=> true
```

**Comment by quartzmo** [comment]
Closed by googleapis/gax-ruby#225
